namedtensor/test_core.py

```diff
@@ -284,7 +284,10 @@ def test_gather():
 
     t = ntorch.tensor(torch.Tensor([[1, 2], [3, 4]]), ("a", "b"))
     index = ntorch.tensor(torch.LongTensor([[0, 0], [1, 0]]), ("a", "c"))
-    ntensor = ntorch.gather(t, "b", index, "c")
+    # Gather will move "b" and "c" to the front for t and index respectively
+    # so we must force the order in order to compare to the original
+    # torch.gather.
+    ntensor = ntorch.gather(t, "b", index, "c")._force_order(("a", "c"))
```

Comment:
This isn't a good unit test. It shouldn't call any _ functions.

Reply:
Since we don't assume any ordering, is the right approach to try all permutations of the output `ntensor` and pass if any of them succeed (equal `base`)?

Reply:
I think the `ntorch.equal` function will now do this automatically. But either way isn't the function you want just `.transpose`? More importantly does this test prove to me that your change works?

Reply:
Great, thanks. I agree that the unit test doesn't test anything, but I wanted to ask about how to compare first. I can write a better test.